lang/rust/avro/src/types.rs

```diff
@@ -3138,4 +3138,40 @@ Field with name '"b"' is not a member of the map items"#,
             )
         );
     }
+
+    #[test]
+    fn avro_4024_resolve_double_from_unknown_string_err() -> TestResult {
+        let schema = Schema::parse_str(r#"{"type": "double"}"#)?;
+        let value = Value::String("unknown".to_owned());
+        match value.resolve(&schema) {
+            Err(err @ Error::GetDouble(_)) => {
+                assert_eq!(
+                    format!("{err:?}"),
+                    r#"Double expected, got String("unknown")"#
+                );
+            }
+            other => {
+                panic!("Expected Error::GetDouble, got {other:?}");
+            }
+        }
```

Comment:
Just FYI: there's a tool that can make testing a lot more happier https://github.com/rust-analyzer/expect-test

Basically it's just like
```rust
assert_eq!(format!("{:?}", value.resolve(&schema)), "balabala")
```
We don't manually `match`, but just print out the result.


What's cooler is that `expect_test` can automatically update the actual result for you.